code/xcvm/cosmwasm/contracts/common/src/gateway.rs

```diff
@@ -1,13 +1,69 @@
-use crate::shared::BridgeMsg;
-use cosmwasm_std::{Addr, CosmosMsg};
+use cw_xc_utils::DefaultXCVMProgram;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
-use xc_core::NetworkId;
+use xc_core::{CallOrigin, Displayed, Funds, InterpreterOrigin, NetworkId};
+
+/// Prefix used for all events attached to gateway responses.
+pub const EVENT_PREFIX: &str = "xcvm.gateway";
+
+/// Version of IBC channels used by the gateway.
+pub const IBC_VERSION: &str = "xcvm-v0";
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+	/// Address of the XCVM registry contract
+	pub registry_address: String,
```

Comment:
My thinking is that because we’re validating this address (i.e. go through `deps.api.addr_validate(&msg.registry_address)`) it’s ‘just a string’ at this point and only once its validated its an `Addr`.  I don’t know if this is a useful way of thinking about it though.